checksum/checksum_test.go

```diff
@@ -0,0 +1,169 @@
+package checksum
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+var testdata = filepath.Join("..", "testdata")
+
+func TestCreate(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		src      string
+		target   string
+		expected int
+		err      bool
+	}{
+		{
+			"Src directory does not exists",
+			"../no-dir",
+			t.TempDir(),
+			0,
+			true,
+		},
+		{
+			"Create checksum files",
+			testdata,
+			t.TempDir(),
+			16,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			err := Create(tc.src, tc.target)
+
+			if !tc.err && err != nil {
+				t.Error("unexpected err")
```

Comment:
Vamos ser mais precisos para ajudar a gente do futuro a debugar?

```suggestion
				t.Errorf("expected no errors creating checksum, for %s", err)
```